### ISSUE TITLE
It seems it does not work now.Output 0

### ISSUE THREAD

**Issue by Lanposter** [walker-1024/furniture_filter#1]
用原始文件跑了代码出来的output是0大小的文件夹，里面什么都没有.
选择的"资源所在文件夹"选了这个项目解压出来的文件夹/OpenArknightssFBS解压出来的文件夹/安卓的AB文件夹都试过了都没有输出东西.是因为给家具命名的方式变了吗?

**Comment by walker-1024** [comment]
看你的描述，似乎是直接使用了安卓手机里的AB文件夹当做“资源所在文件夹”了，那么你缺少了一个叫做解包的步骤，可以参考这个文章 https://www.bilibili.com/read/cv2777112/ 中的第一部分。
我刚把Readme写的更详细了些，你可以再参考下。

**Comment by Lanposter** [comment]
确实。忘记试解包出来的文件了。谢谢。不过，我现在试了解包的文件了，使用的是生路活动上一个版本的游戏的包，为了防止文件过大进度太慢我是去掉了audio文件夹和含有"battle"的文件进行解包的。返回来的结果依然是0.运行完其中一个部分是这样写的：
开始处理 furni_set_s1_02
The length of furnitureIDArray is 15.
找到含家具ID的文件夹0个，找到含主题简名的png文件0个。
找到家具图标0个。
furni_set_s1_02 Done

**Comment by walker-1024** [comment]
是每个主题都没有检索到文件，还是只有个别主题呢。
我更新的Readme中的图片是我昨天使用官网最新下载的apk文件测试的结果，所以至少在我这儿是正常的。
我还是怀疑你的资源文件缺少或者哪里没处理好，不行的话可以发些详细截图，我有空帮你看看。
如果确定资源文件没问题，那可能需要你自己调试代码了。

**Comment by Lanposter** [comment]
之前是每个主题都没有检索到文件。

- 我后来像你一样直接解压apk进行解包再进行检索成功找到了一些文件，我返回去对比我原来的文件发现是热更新后模拟器里的游戏文件没有building这个文件夹导致我后续解包出来也没有相关内容。但是我不清楚apk里是否就已经含有所有家具文件了，我也不知道去哪找热更新后模拟器building文件夹的去向。

- 我在apk解压解包的文件里检索有小半都是0个含家具ID的文件夹，总感觉缺了很多东西，但是最新的家具的模型却都是存在着的，我打开看过了确实是最新的家具模型。

- 我AssetStudio之前提示过我无法使用OpenGL,这会导致解包出来的家具有丢失吗？
- 如果不是主题套件里的家具，单品的家具如何找到对应的那些文件有什么好的建议吗？

- 另外大佬有没有研究过家具的着色器哪些对应哪些用途呢？如果有研究过有什么值得一提的经验可以分享一下吗？

谢谢大佬的解答！附件是对apk直接解压解包（去除audio和battle后解包的）运行过滤器显示出来的过程
[apklog.txt](https://github.com/user-attachments/files/15792775/apklog.txt)

**Comment by walker-1024** [comment]
1&2：确实，我看了下我手机中对应的位置，也没有building文件夹。我当初写的时候，也是会有一些主题拿不到模型，但我并没有很关注完整度。现在想了下，我觉得有几种可能。

1. apk中家具文件不齐全，剩下的由游戏热更新时下载，但下载后放在了某个访问不了的地方，或者并不是ab格式的文件。
2. apk中家具文件是齐全的，但在使用AssetStudio解包时出现了一些错误，导致有一部分没有解包成功。
3. apk中家具文件是齐全的，解包也全部成功了，但游戏中对于家具模型的使用存在两种方案，两种方案对于文件的命名方式或文件格式不同，我只发现了其中一种方案的命名方式，所以只能找到一部分家具模型，另一部分家具模型也存在，只是没有被找到。

我比较倾向于第3种可能。

3：没遇到过，但我觉得不会。

4：这个挺好搞的，themeId为空字符串的就是单品家具，寻找文件的方法其实是一样的，我直接写了下，single.py。当然，也是只能找到一部分家具的模型。

5：没研究过。
